drivers/serial/uart_rpi_pico.c

```diff
@@ -74,12 +75,22 @@ static int uart_rpi_init(const struct device *dev)
 	 * uart_init() may be replaced by register based API once rpi-pico platform
 	 * has a clock controller driver
 	 */
-	baudrate = uart_init(uart_inst, data->baudrate);
+	baudrate = uart_init(uart_inst, data->uart_config.baudrate);
 	/* Check if baudrate adjustment returned by 'uart_init' function is a positive value */
-	if (baudrate <= 0) {
+	if (baudrate == 0) {
 		return -EINVAL;
 	}
-
+	/*
+	 * initialize uart_config with hardware reset values
+	 * https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#tab-registerlist_uart page:431
+	 */
+	data->uart_config = (struct uart_config){
+		.baudrate = baudrate,
+		.data_bits = UART_CFG_DATA_BITS_5,
```

Comment:
5 data bits? Might be the default for the silicon but that would be very non-standard

Reply:
It does seem non standard to me as well.
However according the documentation:


**UARTLCR_H**

Bits | Name | Description | Type | Reset
-- | -- | -- | -- | --
6:5 | WLEN | Word length. These bits indicate the number of data bits transmitted or received in a frame as follows: b11 = 8 bits b10 = 7 bits b01 = 6 bits b00 = 5 bits. | RW | 0x0

https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#tab-registerlist_uart page:431

Reply:
Hm... I think we should set this to 8-bit as default. UARTs are used for console/log output (most of the time). So setting to 5-bit is not going to work.

Reply:
Changed the default to 8 bits